packages/core/src/account/base.ts

```diff
@@ -316,9 +296,36 @@ export abstract class BaseSmartContractAccount<
   protected async parseFactoryAddressFromAccountInitCode(): Promise<
     [Address, Hex]
   > {
-    const initCode = await this.getAccountInitCode();
+    const initCode = await this._getAccountInitCode();
     const factoryAddress = `0x${initCode.substring(2, 42)}` as Address;
     const factoryCalldata = `0x${initCode.substring(42)}` as Hex;
     return [factoryAddress, factoryCalldata];
   }
+
+  private async _getAccountInitCode(): Promise<Hash> {
```

Comment:
why would we not want to use this method publicly, which leverages for the cached override?

Reply:
using this means that all of the methods on base actually depend on `getAccountInitCode` uses the override correctly. 

If instead this change was made to `getAccountInitCode` all implementors of `BaseSmartContractAccount` would have to call `super.getAccountInitCode` and check if that returns something. Thats not much different than just updating all sub account impls to check if `this.accountInitCode` is set. 

With this change, all existing implementations get this functionality for free